packages/grid/x-data-grid/src/models/colDef/gridColDef.ts

```diff
@@ -215,6 +216,19 @@ export interface GridColDef<R extends GridValidRowModel = any, V = any, F = V> {
    * Allows setting the filter operators for this column.
    */
   filterOperators?: GridFilterOperator<R, V, F>[];
+  /**
+   * The callback that generates a filtering function for a given quick filter value.
+   * This function can return `null` to skip filtering for this value and column.
+   * @param {any} value The value with which we want to filter the column.
+   * @param {GridStateColDef} colDef The column from which we want to filter the rows.
+   * @param {React.MutableRefObject<GridApiCommunity>} apiRef The API of the grid.
+   * @returns {null | ((params: GridCellParams) => boolean)} The function to call to check if a row pass this filter value or not.
+   */
+  getApplyQuickFilterFn?: (
+    value: any,
+    colDef: GridStateColDef,
+    apiRef: React.MutableRefObject<GridApiCommunity>,
```

Comment:
Is there a use case where `colDef` and `apiRef` need to be accessed?

`colDef` could be redundant since the function is defined inside the column definition.

Reply:
The code is taken form how the [`GridFilterInputSingleSelect`](https://github.com/alexfauquette/material-ui-x/blob/quick-filter/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputSingleSelect.tsx#L10:L32) renders options

Reply:
> The code is taken form how the [GridFilterInputSingleSelect](https://github.com/alexfauquette/material-ui-x/blob/quick-filter/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputSingleSelect.tsx#L10:L32) renders options

This makes me think that maybe `singleSelect` should have a default value formatter.

Reply:
@m4theushw I added this to #3287